ISM policy creation error in multi-node Data Node startup

## Description

I see the following exception when starting a two-node Data Node setup in Docker Compose. The issue only shows up for one Data Node instance.

```
data-node-2  | 2024-03-27T10:03:33.700Z ERROR [OpensearchNodeHeartbeat] Uncaught exception in Periodical
data-node-2  | org.graylog.shaded.opensearch2.org.opensearch.OpenSearchException: Unable to create ism policy
data-node-2  | 	at org.graylog.storage.opensearch2.OpenSearchClient.exceptionFrom(OpenSearchClient.java:184) ~[graylog-storage-opensearch2-6.0.0-beta.3.jar:?]
data-node-2  | 	at org.graylog.storage.opensearch2.OpenSearchClient.execute(OpenSearchClient.java:135) ~[graylog-storage-opensearch2-6.0.0-beta.3.jar:?]
data-node-2  | 	at org.graylog.storage.opensearch2.ism.IsmApi.perform(IsmApi.java:89) ~[graylog-storage-opensearch2-6.0.0-beta.3.jar:?]
data-node-2  | 	at org.graylog.storage.opensearch2.ism.IsmApi.createPolicy(IsmApi.java:61) ~[graylog-storage-opensearch2-6.0.0-beta.3.jar:?]
data-node-2  | 	at org.graylog.storage.opensearch2.DataStreamAdapterOS2.applyIsmPolicy(DataStreamAdapterOS2.java:169) ~[graylog-storage-opensearch2-6.0.0-beta.3.jar:?]
data-node-2  | 	at org.graylog2.indexer.datastream.DataStreamServiceImpl.createDataStream(DataStreamServiceImpl.java:66) ~[graylog2-server-6.0.0-beta.3.jar:?]
data-node-2  | 	at org.graylog.datanode.metrics.ConfigureMetricsIndexSettings.lambda$transition$0(ConfigureMetricsIndexSettings.java:88) ~[graylog-datanode.jar:?]
data-node-2  | 	at java.base/java.util.Optional.ifPresent(Unknown Source) ~[?:?]
data-node-2  | 	at org.graylog.datanode.metrics.ConfigureMetricsIndexSettings.transition(ConfigureMetricsIndexSettings.java:78) ~[graylog-datanode.jar:?]
data-node-2  | 	at org.graylog.datanode.metrics.ConfigureMetricsIndexSettings.transition(ConfigureMetricsIndexSettings.java:52) ~[graylog-datanode.jar:?]
data-node-2  | 	at org.graylog.datanode.management.StateMachineTracerAggregator.lambda$transition$1(StateMachineTracerAggregator.java:45) ~[graylog-datanode.jar:?]
data-node-2  | 	at java.base/java.lang.Iterable.forEach(Unknown Source) ~[?:?]
data-node-2  | 	at org.graylog.datanode.management.StateMachineTracerAggregator.transition(StateMachineTracerAggregator.java:45) ~[graylog-datanode.jar:?]
data-node-2  | 	at org.graylog.datanode.management.StateMachineTracerAggregator.transition(StateMachineTracerAggregator.java:26) ~[graylog-datanode.jar:?]
data-node-2  | 	at com.github.oxo42.stateless4j.StateMachine.publicFire(StateMachine.java:227) ~[stateless4j-2.6.0.jar:?]
data-node-2  | 	at com.github.oxo42.stateless4j.StateMachine.fire(StateMachine.java:145) ~[stateless4j-2.6.0.jar:?]
data-node-2  | 	at org.graylog.datanode.management.OpensearchProcessImpl.onEvent(OpensearchProcessImpl.java:149) ~[graylog-datanode.jar:?]
data-node-2  | 	at org.graylog.datanode.periodicals.OpensearchNodeHeartbeat.onNodeResponse(OpensearchNodeHeartbeat.java:69) ~[graylog-datanode.jar:?]
data-node-2  | 	at org.graylog.datanode.periodicals.OpensearchNodeHeartbeat.doRun(OpensearchNodeHeartbeat.java:60) ~[graylog-datanode.jar:?]
data-node-2  | 	at org.graylog2.plugin.periodical.Periodical.run(Periodical.java:99) [graylog2-server-6.0.0-beta.3.jar:?]
data-node-2  | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
data-node-2  | 	at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source) [?:?]
data-node-2  | 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?]
data-node-2  | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
data-node-2  | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
data-node-2  | 	at java.base/java.lang.Thread.run(Unknown Source) [?:?]
data-node-2  | Caused by: org.graylog.shaded.opensearch2.org.opensearch.client.ResponseException: method [PUT], host [http://data-node-2:9200], URI [/_plugins/_ism/policies/gl-datanode-metrics-ism?format=json], status line [HTTP/1.1 409 Conflict]
data-node-2  | {"error":{"root_cause":[{"type":"version_conflict_engine_exception","reason":"[gl-datanode-metrics-ism]: version conflict, document already exists (current version [1])","index":".opendistro-ism-config","shard":"0","index_uuid":"OZSexHSlQ12ENN0H9jRj6g"}],"type":"version_conflict_engine_exception","reason":"[gl-datanode-metrics-ism]: version conflict, document already exists (current version [1])","index":".opendistro-ism-config","shard":"0","index_uuid":"OZSexHSlQ12ENN0H9jRj6g"},"status":409}
data-node-2  | 	at org.graylog.shaded.opensearch2.org.opensearch.client.RestClient.convertResponse(RestClient.java:376) ~[graylog-storage-opensearch2-6.0.0-beta.3.jar:?]
data-node-2  | 	at org.graylog.shaded.opensearch2.org.opensearch.client.RestClient.performRequest(RestClient.java:346) ~[graylog-storage-opensearch2-6.0.0-beta.3.jar:?]
data-node-2  | 	at org.graylog.shaded.opensearch2.org.opensearch.client.RestClient.performRequest(RestClient.java:321) ~[graylog-storage-opensearch2-6.0.0-beta.3.jar:?]
data-node-2  | 	at org.graylog.storage.opensearch2.ism.IsmApi.lambda$perform$0(IsmApi.java:94) ~[graylog-storage-opensearch2-6.0.0-beta.3.jar:?]
data-node-2  | 	at org.graylog.storage.opensearch2.OpenSearchClient.execute(OpenSearchClient.java:133) ~[graylog-storage-opensearch2-6.0.0-beta.3.jar:?]
data-node-2  | 	... 24 more
```

While testing the Docker Compose file again for this issue, I noticed a different error:

```
data-node-1  | 2024-03-27T10:07:19.632Z INFO  [OpensearchProcessImpl] [2024-03-27T10:07:19,629][WARN ][r.suppressed             ] [data-node-1] path: /_plugins/_ism/policies/gl-datanode-metrics-ism, params: {policyID=gl-datanode-metrics-ism, format=json}
data-node-1  | 2024-03-27T10:07:19.633Z INFO  [OpensearchProcessImpl] org.opensearch.action.NoShardAvailableActionException: No shard available for [get [.opendistro-ism-config][gl-datanode-metrics-ism]: routing [null]]
data-node-1  | 2024-03-27T10:07:19.633Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.perform(TransportSingleShardAction.java:256) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.633Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.onFailure(TransportSingleShardAction.java:239) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.633Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleException(TransportSingleShardAction.java:303) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.633Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1511) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.633Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundHandler.lambda$handleException$5(InboundHandler.java:447) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.633Z INFO  [OpensearchProcessImpl] 	at org.opensearch.common.util.concurrent.OpenSearchExecutors$DirectExecutorService.execute(OpenSearchExecutors.java:343) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.633Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundHandler.handleException(InboundHandler.java:445) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundHandler.handlerResponseError(InboundHandler.java:437) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundHandler.messageReceived(InboundHandler.java:170) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundHandler.inboundMessage(InboundHandler.java:127) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.TcpTransport.inboundMessage(TcpTransport.java:770) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:175) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:150) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:115) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:95) [transport-netty4-client-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.634Z INFO  [OpensearchProcessImpl] 	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:280) [netty-handler-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.635Z INFO  [OpensearchProcessImpl] 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) [netty-common-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] 	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] Caused by: org.opensearch.transport.RemoteTransportException: [data-node-2][10.0.19.4:9300][indices:data/read/get[s]]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] Caused by: org.opensearch.index.shard.IllegalIndexShardStateException: CurrentState[RECOVERING] operations only allowed when shard state is one of [POST_RECOVERY, STARTED]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] 	at org.opensearch.index.shard.IndexShard.readAllowed(IndexShard.java:2610) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] 	at org.opensearch.index.shard.IndexShard.get(IndexShard.java:1332) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] 	at org.opensearch.index.get.ShardGetService.innerGet(ShardGetService.java:229) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] 	at org.opensearch.index.get.ShardGetService.get(ShardGetService.java:142) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] 	at org.opensearch.index.get.ShardGetService.get(ShardGetService.java:126) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.get.TransportGetAction.shardOperation(TransportGetAction.java:156) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.get.TransportGetAction.shardOperation(TransportGetAction.java:62) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.636Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.support.single.shard.TransportSingleShardAction.lambda$asyncShardOperation$0(TransportSingleShardAction.java:131) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.637Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:74) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.637Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.ActionRunnable$2.doRun(ActionRunnable.java:89) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.637Z INFO  [OpensearchProcessImpl] 	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:913) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.637Z INFO  [OpensearchProcessImpl] 	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.637Z INFO  [OpensearchProcessImpl] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
data-node-1  | 2024-03-27T10:07:19.637Z INFO  [OpensearchProcessImpl] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
data-node-1  | 2024-03-27T10:07:19.637Z INFO  [OpensearchProcessImpl] 	at java.lang.Thread.run(Thread.java:1583) ~[?:?]
data-node-1  | 2024-03-27T10:07:19.661Z INFO  [OpensearchProcessImpl] [2024-03-27T10:07:19,660][ERROR][o.o.t.InboundHandler     ] [data-node-1] failed to handle exception response [org.opensearch.transport.TransportService$ContextRestoreResponseHandler/org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2@41f241d7]
data-node-1  | 2024-03-27T10:07:19.662Z INFO  [OpensearchProcessImpl] java.lang.NullPointerException: Cannot invoke "org.opensearch.action.get.GetResponse.isExists()" because the return value of "org.opensearch.action.get.MultiGetItemResponse.getResponse()" is null
data-node-1  | 2024-03-27T10:07:19.662Z INFO  [OpensearchProcessImpl] 	at org.opensearch.indexmanagement.indexstatemanagement.transport.action.removepolicy.TransportRemovePolicyAction$RemovePolicyHandler$getExistingManagedIndices$2.onResponse(TransportRemovePolicyAction.kt:225) ~[?:?]
data-node-1  | 2024-03-27T10:07:19.662Z INFO  [OpensearchProcessImpl] 	at org.opensearch.indexmanagement.indexstatemanagement.transport.action.removepolicy.TransportRemovePolicyAction$RemovePolicyHandler$getExistingManagedIndices$2.onResponse(TransportRemovePolicyAction.kt:206) ~[?:?]
data-node-1  | 2024-03-27T10:07:19.662Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:113) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.662Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:107) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.662Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.get.TransportMultiGetAction$1.finishHim(TransportMultiGetAction.java:168) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.662Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.get.TransportMultiGetAction$1.onFailure(TransportMultiGetAction.java:163) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.662Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.support.TransportAction$1.onFailure(TransportAction.java:122) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.662Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.perform(TransportSingleShardAction.java:262) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.662Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.onFailure(TransportSingleShardAction.java:239) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.663Z INFO  [OpensearchProcessImpl] 	at org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleException(TransportSingleShardAction.java:303) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.663Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1511) ~[opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.663Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundHandler.lambda$handleException$5(InboundHandler.java:447) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.663Z INFO  [OpensearchProcessImpl] 	at org.opensearch.common.util.concurrent.OpenSearchExecutors$DirectExecutorService.execute(OpenSearchExecutors.java:343) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.663Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundHandler.handleException(InboundHandler.java:445) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.663Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundHandler.handlerResponseError(InboundHandler.java:437) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.663Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundHandler.messageReceived(InboundHandler.java:170) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.663Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundHandler.inboundMessage(InboundHandler.java:127) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.663Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.TcpTransport.inboundMessage(TcpTransport.java:770) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.663Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:175) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.663Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:150) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.664Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:115) [opensearch-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.664Z INFO  [OpensearchProcessImpl] 	at org.opensearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:95) [transport-netty4-client-2.12.0.jar:2.12.0]
data-node-1  | 2024-03-27T10:07:19.664Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.664Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.664Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.664Z INFO  [OpensearchProcessImpl] 	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:280) [netty-handler-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.664Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.664Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.664Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.664Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.665Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.665Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.665Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.665Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.665Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.665Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.665Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.665Z INFO  [OpensearchProcessImpl] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) [netty-transport-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.665Z INFO  [OpensearchProcessImpl] 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) [netty-common-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.665Z INFO  [OpensearchProcessImpl] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.106.Final.jar:4.1.106.Final]
data-node-1  | 2024-03-27T10:07:19.665Z INFO  [OpensearchProcessImpl] 	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
```

## Steps to Reproduce (for bugs)

1. Start the following Docker Compose setup with `docker compose up`
2. Observe the error in the logs

```yaml
x-data-node-env: &data-node-env
  GRAYLOG_DATANODE_NODE_ID_FILE: "/var/lib/graylog-datanode/node-id"
  GRAYLOG_DATANODE_PASSWORD_SECRET: "${GRAYLOG_PASSWORD_SECRET:?Please configure GRAYLOG_PASSWORD_SECRET in the .env file}"
  GRAYLOG_DATANODE_ROOT_PASSWORD_SHA2: "${GRAYLOG_ROOT_PASSWORD_SHA2:?Please configure GRAYLOG_ROOT_PASSWORD_SHA2 in the .env file}"
  GRAYLOG_DATANODE_MONGODB_URI: "mongodb://mongodb:27017/graylog"
  GRAYLOG_DATANODE_INSECURE_STARTUP: "true"
  GRAYLOG_DATANODE_NODE_SEARCH_CACHE_SIZE: "10g"
  GRAYLOG_DATANODE_S3_CLIENT_DEFAULT_ACCESS_KEY: "minioadmin"
  GRAYLOG_DATANODE_S3_CLIENT_DEFAULT_SECRET_KEY: "minioadmin"
  GRAYLOG_DATANODE_S3_CLIENT_DEFAULT_ENDPOINT: "minio:9000"
  GRAYLOG_DATANODE_S3_CLIENT_DEFAULT_PROTOCOL: "http"
  GRAYLOG_DATANODE_S3_CLIENT_DEFAULT_REGION: "us-east-1"
  GRAYLOG_DATANODE_S3_CLIENT_DEFAULT_PATH_STYLE_ACCESS: "true"
  GRAYLOG_DATANODE_OPENSEARCH_PLUGINS_LOCATION: "/usr/share/graylog-datanode/dist/plugins"

x-data-node-ulimits: &data-node-ulimits
  memlock:
    hard: -1
    soft: -1
  nofile:
    soft: 65536
    hard: 65536

services:
  mongodb:
    container_name: "mongodb"
    image: "mongo:7.0"
    volumes:
      - "mongodb:/data/db"
    restart: "on-failure"

  data-node-1:
    container_name: "data-node-1"
    image: "graylog/graylog-datanode:6.0.0-beta.3-1"
    hostname: "data-node-1"
    environment:
      <<: *data-node-env
    ulimits:
      <<: *data-node-ulimits
    volumes:
      - "graylog-data-node-1:/var/lib/graylog-datanode"
    restart: "on-failure"

  data-node-2:
    container_name: "data-node-2"
    image: "graylog/graylog-datanode:6.0.0-beta.3-1"
    hostname: "data-node-2"
    environment:
      <<: *data-node-env
    ulimits:
      <<: *data-node-ulimits
    volumes:
      - "graylog-data-node-2:/var/lib/graylog-datanode"
    restart: "on-failure"

volumes:
  mongodb:
  graylog-data-node-1:
  graylog-data-node-2:
```

## Your Environment

* Graylog Version: 6.0.0-beta.3

--- a/changelog/unreleased/pr-19241.toml
+++ b/changelog/unreleased/pr-19241.toml
@@ -1,0 +1,5 @@
+type = "fixed"
+message = "Fix data stream replicas and creation on single/multi-node data node setups."
+
+issues = ["18830,19010"]
+pulls = ["19241"]

--- a/changelog/unreleased/pr-19241.toml
+++ b/changelog/unreleased/pr-19241.toml
@@ -1,5 +1,5 @@
 type = "fixed"
 message = "Fix data stream replicas and creation on single/multi-node data node setups."
 
-issues = ["18830,19010"]
+issues = ["18830", "19010"]
 pulls = ["19241"]

--- a/data-node/src/main/java/org/graylog/datanode/metrics/ConfigureMetricsIndexSettings.java
+++ b/data-node/src/main/java/org/graylog/datanode/metrics/ConfigureMetricsIndexSettings.java
@@ -74,7 +74,7 @@ public class ConfigureMetricsIndexSettings implements StateMachineTracer {
 
     @Override
     public void transition(OpensearchEvent trigger, OpensearchState source, OpensearchState destination) {
-        if (destination == OpensearchState.AVAILABLE && source == OpensearchState.STARTING) {
+        if (destination == OpensearchState.AVAILABLE && source == OpensearchState.STARTING && process.isManagerNode()) {
             process.openSearchClient().ifPresent(client -> {
                 final IsmApi ismApi = new IsmApi(client, objectMapper);
                 int replicas = nodeService.allActive().size() == 1 ? 0 : 1;

--- a/data-node/src/main/java/org/graylog/datanode/opensearch/OpensearchProcess.java
+++ b/data-node/src/main/java/org/graylog/datanode/opensearch/OpensearchProcess.java
@@ -45,4 +45,6 @@ public interface OpensearchProcess extends ManagableProcess<OpensearchConfigurat
 
     void onReset();
 
+    boolean isManagerNode();
+
 }

--- a/data-node/src/main/java/org/graylog/datanode/opensearch/OpensearchProcessImpl.java
+++ b/data-node/src/main/java/org/graylog/datanode/opensearch/OpensearchProcessImpl.java
@@ -31,8 +31,11 @@ import org.graylog.datanode.opensearch.statemachine.OpensearchEvent;
 import org.graylog.datanode.opensearch.statemachine.OpensearchState;
 import org.graylog.datanode.opensearch.statemachine.OpensearchStateMachine;
 import org.graylog.datanode.opensearch.statemachine.tracer.StateMachineTracer;
+import org.graylog.datanode.periodicals.ClusterStateResponse;
 import org.graylog.datanode.process.ProcessInformation;
 import org.graylog.datanode.process.ProcessListener;
+import org.graylog.shaded.opensearch2.org.opensearch.client.Request;
+import org.graylog.shaded.opensearch2.org.opensearch.client.Response;
 import org.graylog.shaded.opensearch2.org.opensearch.client.RestHighLevelClient;
 import org.graylog.storage.opensearch2.OpenSearchClient;
 import org.graylog2.cluster.nodes.DataNodeDto;
@@ -283,4 +286,25 @@ class OpensearchProcessImpl implements OpensearchProcess, ProcessListener {
         LOG.warn("Opensearch process failed", e);
         onEvent(OpensearchEvent.PROCESS_TERMINATED);
     }
+
+    @Override
+    public boolean isManagerNode() {
+        return restClient()
+                .flatMap(this::requestClusterState)
+                .map(r -> r.nodes().get(r.clusterManagerNode()))
+                .map(managerNode -> configuration.getDatanodeNodeName().equals(managerNode.name()))
+                .orElse(false);
+    }
+    
+    private Optional<ClusterStateResponse> requestClusterState(RestHighLevelClient client) {
+        try {
+            final Response response = client.getLowLevelClient().performRequest(new Request("GET", "_cluster/state/"));
+            final ClusterStateResponse state = objectMapper.readValue(response.getEntity().getContent(), ClusterStateResponse.class);
+            return Optional.of(state);
+        } catch (IOException e) {
+            LOG.warn("Failed to obtain cluster state response", e);
+            return Optional.empty();
+        }
+    }
+
 }

--- a/data-node/src/main/java/org/graylog/datanode/periodicals/MetricsCollector.java
+++ b/data-node/src/main/java/org/graylog/datanode/periodicals/MetricsCollector.java
@@ -31,9 +31,7 @@ import org.graylog.shaded.opensearch2.org.opensearch.action.index.IndexRequest;
 import org.graylog.shaded.opensearch2.org.opensearch.action.index.IndexResponse;
 import org.graylog.shaded.opensearch2.org.opensearch.action.search.SearchRequest;
 import org.graylog.shaded.opensearch2.org.opensearch.action.search.SearchResponse;
-import org.graylog.shaded.opensearch2.org.opensearch.client.Request;
 import org.graylog.shaded.opensearch2.org.opensearch.client.RequestOptions;
-import org.graylog.shaded.opensearch2.org.opensearch.client.Response;
 import org.graylog.shaded.opensearch2.org.opensearch.client.RestHighLevelClient;
 import org.graylog.shaded.opensearch2.org.opensearch.core.action.ActionListener;
 import org.graylog.shaded.opensearch2.org.opensearch.index.query.QueryBuilders;
@@ -52,7 +50,6 @@ import java.lang.management.MemoryUsage;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
-import java.util.Optional;
 
 public class MetricsCollector extends Periodical {
 
@@ -122,33 +119,13 @@ public class MetricsCollector extends Periodical {
                 indexRequest.source(metrics);
                 indexDocument(client, indexRequest);
 
-                if (isManagerNode(process)) {
+                if (process.isManagerNode()) {
                     metrics = new HashMap<>(clusterStatMetricsCollector.getClusterMetrics(getPreviousMetricsForCluster(client)));
                     metrics.put(configuration.getMetricsTimestamp(), new DateTime(DateTimeZone.UTC));
                     indexRequest.source(metrics);
                     indexDocument(client, indexRequest);
                 }
             });
-        }
-    }
-
-    public boolean isManagerNode(OpensearchProcess process) {
-        return process.restClient()
-                .flatMap(this::requestClusterState)
-                .map(r -> r.nodes().get(r.clusterManagerNode()))
-                .map(managerNode -> configuration.getDatanodeNodeName().equals(managerNode.name()))
-                .orElse(false);
-    }
-
-
-    private Optional<ClusterStateResponse> requestClusterState(RestHighLevelClient client) {
-        try {
-            final Response response = client.getLowLevelClient().performRequest(new Request("GET", "_cluster/state/"));
-            final ClusterStateResponse state = objectMapper.readValue(response.getEntity().getContent(), ClusterStateResponse.class);
-            return Optional.of(state);
-        } catch (IOException e) {
-            LOG.warn("Failed to obtain cluster state response", e);
-            return Optional.empty();
         }
     }
 

--- a/graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/DataStreamAdapterOS2.java
+++ b/graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/DataStreamAdapterOS2.java
@@ -19,6 +19,7 @@ package org.graylog.storage.opensearch2;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import jakarta.inject.Inject;
 import org.graylog.shaded.opensearch2.org.opensearch.OpenSearchException;
+import org.graylog.shaded.opensearch2.org.opensearch.action.admin.cluster.settings.ClusterUpdateSettingsRequest;
 import org.graylog.shaded.opensearch2.org.opensearch.action.admin.indices.settings.put.UpdateSettingsRequest;
 import org.graylog.shaded.opensearch2.org.opensearch.action.support.master.AcknowledgedResponse;
 import org.graylog.shaded.opensearch2.org.opensearch.client.indices.CreateDataStreamRequest;
@@ -27,9 +28,11 @@ import org.graylog.shaded.opensearch2.org.opensearch.client.indices.DeleteDataSt
 import org.graylog.shaded.opensearch2.org.opensearch.client.indices.GetDataStreamRequest;
 import org.graylog.shaded.opensearch2.org.opensearch.client.indices.GetDataStreamResponse;
 import org.graylog.shaded.opensearch2.org.opensearch.client.indices.PutComposableIndexTemplateRequest;
+import org.graylog.shaded.opensearch2.org.opensearch.client.indices.PutIndexTemplateRequest;
 import org.graylog.shaded.opensearch2.org.opensearch.cluster.metadata.ComposableIndexTemplate;
 import org.graylog.shaded.opensearch2.org.opensearch.cluster.metadata.DataStream;
 import org.graylog.shaded.opensearch2.org.opensearch.common.compress.CompressedXContent;
+import org.graylog.shaded.opensearch2.org.opensearch.common.settings.Settings;
 import org.graylog.storage.opensearch2.ism.IsmApi;
 import org.graylog2.indexer.datastream.DataStreamAdapter;
 import org.graylog2.indexer.datastream.Policy;
@@ -123,9 +126,27 @@ public class DataStreamAdapterOS2 implements DataStreamAdapter {
         try {
             client.execute((c, requestOptions) -> c.indices().putSettings(req, requestOptions));
         } catch (Exception e) {
-            log.debug("Could not set replicas for .opendistro-job-scheduler-locl system index. It might not exist yet.");
+            log.debug("Could not set replicas for .opendistro-job-scheduler-lock system index. It might not exist yet.");
         }
         req.indices(dataStreamName);
+
+        PutIndexTemplateRequest templateRequest = new PutIndexTemplateRequest("data-stream-system-indices");
+        templateRequest.patterns(List.of(".opendistro-job-scheduler-lock"));
+        templateRequest.settings(Settings.builder().put("number_of_replicas", replicas).build());
+        try {
+            client.execute((c, requestOptions) -> c.indices().putTemplate(templateRequest, requestOptions));
+        } catch (Exception e) {
+            log.debug("Could not set replicas for data stream system indices");
+        }
+
+        ClusterUpdateSettingsRequest settingsRequest = new ClusterUpdateSettingsRequest();
+        settingsRequest.persistentSettings(Settings.builder().put("opendistro.index_state_management.history.number_of_replicas", replicas).build());
+        try {
+            client.execute((c, requestOptions) -> c.cluster().putSettings(settingsRequest, requestOptions));
+        } catch (Exception e) {
+            log.debug("Could not set default replicas for ism history");
+        }
+
         try {
             client.execute((c, requestOptions) -> c.indices().putSettings(req, requestOptions));
         } catch (Exception e) {

